### ISSUE TITLE
Add Rubinius 4

### ISSUE THREAD

**Issue by pkuczynski** [eregon/use-ruby-action#5]

**Comment by eregon** [comment]
Could you try locally if Rubinius works for your project first?

Which version would you like? It seems there are releases almost every day, that's too many to support: https://github.com/rubinius/rubinius/releases

**Comment by eregon** [comment]
https://github.com/rubinius/rubinius#the-ruby-programming-language says
> Many popular Ruby applications, like Rails, run on Rubinius, which aims to be compatible with Ruby version 2.3.1.

So I don't have much hope for compatibility with recent Ruby code.

**Comment by pkuczynski** [comment]
I guess it's fine to update once in a while. You can also integrate with https://github.com/postmodern/ruby-versions or even initiate build whenever their repo changes - I think that's possible with GHA?

**Comment by eregon** [comment]
I think GitHub Actions doesn't actually allow to react on events from another repository.
Anyway, my point is I don't want to keep 100 releases of Rubinius.

I can try to build the latest release, but it would be helpful if you can assess the compatibility for your use case. If it's really difficult to build and not compatible enough, then I might rethink this.

**Comment by eregon** [comment]
The first issue is `ruby-install` is no longer able to install Rubinius:
https://github.com/eregon/ruby-install-builder/runs/380297030
Also, the dependencies in `ruby-install` for Rubinius are outdated.

**Comment by eregon** [comment]
And `ruby-build` doesn't have files for Rubinius >= 4:
https://github.com/rbenv/ruby-build/tree/master/share/ruby-build

That's pretty much a showstopper for me.
At least one of these two installers should get support for Rubinius.

**Comment by eregon** [comment]
Out of curiosity, can RVM build Rubinius 4?

**Comment by eregon** [comment]
They also have builds at http://rubinius-binaries-rubinius-com.s3-us-west-2.amazonaws.com/index.txt, but apparently not for Ubuntu 18.04, which I think is the most used platform in GitHub Actions.

**Comment by eregon** [comment]
I managed to get the build started from source, let's see how it goes:
https://github.com/eregon/ruby-install-builder/runs/380326221

**Comment by eregon** [comment]
It failed:
```
3: CXX machine/vm.cpp
3: CXX machine/vm_thread_state.cpp
Build time: 382.5293304 seconds
1: LD machine/vm
GEN codedb/platform.conf
rake aborted!
Don't know how to build task '' (see --tasks)

Tasks: TOP => default => spec => build => build:build => core:build => codedb:extensions
(See full trace by running task with --trace)
##[error]Process completed with exit code 1.
```

**Comment by pkuczynski** [comment]
RVM should be able to build Rubinius and if there is an error, we are happy to fix it.

**Comment by eregon** [comment]
RVM-built Rubies contain RVM-specific paths, config and tweaks though, so I don't think I could use it here. I'll file an issue to Rubinius.

**Comment by eregon** [comment]
https://github.com/rubinius/rubinius/issues/3828

**Comment by pkuczynski** [comment]
Cool, thanks!

**Comment by eregon** [comment]
I got now a working build of Rubinius on Ubuntu 18.04: https://github.com/eregon/use-ruby-action/runs/408718396

On Ubuntu 16.04, it segfaults for `bundle install`: https://github.com/eregon/use-ruby-action/runs/408718394
And on macOS Rubinius SEGV while building: https://github.com/rubinius/rubinius/issues/3830

As a note, latest JSON 2.3.0 fails to build on Rubinius.

**Comment by eregon** [comment]
Fixed by 689f94552b57cf6a8e6aa44b0c9ed93bd7a9bbf5, you can now use `ruby-version: rubinius`.

I'm hesitating to document it under `Supported Versions` since it has so many limitations (only works on `ubuntu-latest`, not compatible with recent, Ruby 2.3-like compatibility, etc).

**Comment by eregon** [comment]
Added to the README and made a release: https://github.com/eregon/use-ruby-action/releases/tag/v1.10.0
